weld/llvm.rs

```diff
@@ -806,53 +837,98 @@ impl LlvmGenerator {
                                  par_for: &ParallelForData,
                                  func: &SirFunction,
                                  ctx: &mut FunctionContext) -> WeldResult<()> {
+
         if !(par_for.data.len() == 1 && par_for.data[0].start.is_none()) {
             for iter in par_for.data.iter() {
-                let (data_ll_ty, data_ll_sym) = self.llvm_type_and_name(func, &iter.data)?;
-                let data_prefix = llvm_prefix(&data_ll_ty);
+                if iter.kind == IterKind::NdIter {
+                    /* Note: here we can't use num_iters_str as a proxy for end as the array may not be
+                     * contiguous */
+                    let data_llvm_info = self.get_array_llvm_info(func, ctx, &iter.data, "".to_string(), false)?;
+                    /* int offset = 0;
+                     * for (i = 0; i < len(shapes); i++) {
+                     *     int max_i = shapes[i] - 1;
+                     *     int stride_i = strides[i];
+                     *     offset += max_i*stride_i;
+                     * }
+                     * int max_val = start + offset;
+                     * cmp max_val, data_size_ll_tmp
+                     */
+                    let el_ty = self.llvm_type(&Scalar(I64))?;
+                    /* both these have same type */
+                    let shapes_llvm_info = self.get_array_llvm_info(func, ctx, iter.shapes.as_ref().unwrap(), el_ty.clone(), false)?;
+                    let strides_llvm_info = self.get_array_llvm_info(func, ctx, iter.strides.as_ref().unwrap(), el_ty, false)?;
+                    let offset_ptr = ctx.var_ids.next();
+                    ctx.code.add(format!("{} = alloca i64", offset_ptr));
+                    ctx.code.add(format!("store i64 0, i64* {}", offset_ptr));
+                    let loop_name = "boundscheck_loop";
```

Comment:
How are you currently executing these nested loops? This looks something like a while loop with some sort of `next()` method to get the next index to process on?

Reply:
I was just using a generic template in add_llvm_for_loop to generate the loop.start and loop.end code which handled incrementing the iteration variable etc. But I had a bug in this one - If there were multiple nditers, as in zip(nditer(e0,....), nditer(e1,....)) -- then there would be multiple for loops generated with the same name in llvm. I fixed this with the latest commit. Is this what you were referring to?